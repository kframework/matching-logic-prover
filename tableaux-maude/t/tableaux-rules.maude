in "../tableaux.maude" .

--- No inf traces, satisfiable
search tableaux(\and([ a  (Y{0}) ], c)) =>* Sequent:Sequent .
show search graph .

--- No inf traces, satisfiable
search tableaux(\nu Y . \and([ a  (Y{0}) ], c)) =>* Sequent:Sequent .
show search graph .

--- nu trace, satisfiable
search tableaux(\nu Y . \and(< a (Y{0}) >, c)) =>* Sequent:Sequent .
show search graph .

--- No inf traces, satisfiable
search tableaux(\mu Y . \and([ a  (Y{0}) ], c)) =>* Sequent:Sequent .
show search graph .

--- mu trace, unsatisfiable
search tableaux(\mu Y . \and(< a (Y{0}) >, c)) =>* Sequent:Sequent .
show search graph .

--- satisfiable
--- 1. nu & mu generate infinitely often, but nu is older
--- 2. mu generates infinitely often 

search tableaux(\nu Y . \and(< a (\mu Y . \and(< a  (Y{0}) >, [ a  (Y{1}) ])) > , c)) =>* Sequent:Sequent .

--- tableaux( \and( \mu X . \or ([ a ] X {0}, \nu Y . \and([ a ] Y{0}, c))
---                      , \nu X . \and(< a > X{0}, \mu Y . \or (< a > Y{0}, \not c))
---         )     )
