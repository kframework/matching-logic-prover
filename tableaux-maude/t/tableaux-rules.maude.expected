==========================================
search in MATCHING-LOGIC-TABLEAUX : tableaux(\and(c, [a (Y{0})])) =>* Sequent:Sequent .

Solution 1 (state 0)
Sequent:Sequent --> < c, [a (Y{0})] ; nil >

No more solutions.
state 0, Sequent: < c, [a (Y{0})] ; nil >
==========================================
search in MATCHING-LOGIC-TABLEAUX : tableaux(\nu Y . \and(c, [a (Y{0})])) =>* Sequent:Sequent .

Solution 1 (state 0)
Sequent:Sequent --> < c, [a (v(0){0})] ; v(0) := \nu * . \and(c, [a (*{0})]) >

No more solutions.
state 0, Sequent: < c, [a (v(0){0})] ; v(0) := \nu * . \and(c, [a (*{0})]) >
==========================================
search in MATCHING-LOGIC-TABLEAUX : tableaux(\nu Y . \and(c, < a (Y{0}) >)) =>* Sequent:Sequent .

Solution 1 (state 0)
Sequent:Sequent --> < c, < a (v(0){0}) > ; v(0) := \nu * . \and(c, < a (*{0}) >) >

No more solutions.
state 0, Sequent: < c, < a (v(0){0}) > ; v(0) := \nu * . \and(c, < a (*{0}) >) >
arc 0 ===> state 0 (crl < Gamma, < Sy P > ; DefList > => [P, getBetas(Sy ; Gamma) ; DefList] if isReduced(Gamma ; DefList) = true [label all<>] .)
==========================================
search in MATCHING-LOGIC-TABLEAUX : tableaux(\mu Y . \and(c, [a (Y{0})])) =>* Sequent:Sequent .

Solution 1 (state 0)
Sequent:Sequent --> < c, [a (u(0){0})] ; u(0) := \mu * . \and(c, [a (*{0})]) >

No more solutions.
state 0, Sequent: < c, [a (u(0){0})] ; u(0) := \mu * . \and(c, [a (*{0})]) >
==========================================
search in MATCHING-LOGIC-TABLEAUX : tableaux(\mu Y . \and(c, < a (Y{0}) >)) =>* Sequent:Sequent .

Solution 1 (state 0)
Sequent:Sequent --> < c, < a (u(0){0}) > ; u(0) := \mu * . \and(c, < a (*{0}) >) >

No more solutions.
state 0, Sequent: < c, < a (u(0){0}) > ; u(0) := \mu * . \and(c, < a (*{0}) >) >
arc 0 ===> state 0 (crl < Gamma, < Sy P > ; DefList > => [P, getBetas(Sy ; Gamma) ; DefList] if isReduced(Gamma ; DefList) = true [label all<>] .)
==========================================
search in MATCHING-LOGIC-TABLEAUX : tableaux(\nu Y . \and(c, < a (\mu Y . \and(< a (Y{0}) >, [a (Y{1})])) >)) =>* Sequent:Sequent .

Solution 1 (state 0)
Sequent:Sequent --> < c, < a (\mu * . \and(< a (*{0}) >, [a (v(0){0})])) > ; (v(0) := \nu * . \and(c, < a (\mu * . \and(< a (*{0}) >, [a (*{1})])) >)) (u(1) := \mu * . \and(< a (*{0}) >, [a (v(0){0})])) >

Solution 2 (state 1)
Sequent:Sequent --> < < a (u(1){0}) >, [a (v(0){0})] ; (v(0) := \nu * . \and(c, < a (\mu * . \and(< a (*{0}) >, [a (*{1})])) >)) (u(1) := \mu * . \and(< a (*{0}) >, [a (v(0){0})])) >

Solution 3 (state 2)
Sequent:Sequent --> < c, < a (u(1){0}) >, < a (\mu * . \and(< a (*{0}) >, [a (v(0){0})])) >, [a (v(0){0})] ; (v(0) := \nu * . \and(c, < a (\mu * . \and(< a (*{0}) >, [a (*{1})])) >)) (u(1) := \mu * . \and(< a (*{0}) >, [a (v(0){0})])) >

No more solutions.
Bye.
